PhysicsTools/PatAlgos/plugins/PATPackedCandidateProducer.cc

```diff
@@ -288,10 +294,16 @@ void pat::PATPackedCandidateProducer::produce(edm::StreamID, edm::Event& iEvent,
           outPtrP->back().setAssociationQuality(pat::PackedCandidate::PVAssociationQuality(pat::PackedCandidate::UsedInFitTight));
         }
 
-	// neutrals
+	// neutrals and isolated charged hadrons
+
+        bool isIsolatedChargedHadron = ((cand.pt()>minPtForChargedHadronProperties_)&&(chargedHadronIsolation[reco::PFCandidateRef(cands,ic)]));
+        outPtrP->back().setIsIsolatedChargedHadron(isIsolatedChargedHadron);
 
 	if(abs(cand.pdgId()) == 1 || abs(cand.pdgId()) == 130) {
 	  outPtrP->back().setHcalFraction(cand.hcalEnergy()/(cand.ecalEnergy()+cand.hcalEnergy()));
+        } else if(isIsolatedChargedHadron) {
+          outPtrP->back().setRawCaloFraction((cand.rawEcalEnergy()+cand.rawHcalEnergy())/cand.energy());
```

Comment:
this is not necessarily bound by 1 from above.
Current choice of int8_t for packing with scaling by 100 leads to overflows for the frac > 1.28 which show up as negative values.

The packed representation should probably be changed to uint8_t and in addition some protection should be added either here or in the setRawCaloFraction implementation.

It may even be useful to define an overflow value